src/server/JasmineGraphInstanceService.cpp

```diff
@@ -80,4301 +144,5243 @@ void *instanceservicesession(void *dummyPt) {
-            string size = (data);
-            instance_logger.log("Received file size in bytes: " + size, "info");
-            write(connFd, JasmineGraphInstanceProtocol::SEND_FILE_CONT.c_str(),
-                  JasmineGraphInstanceProtocol::SEND_FILE_CONT.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::SEND_FILE_CONT, "info");
-            string fullFilePath =
-                Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder") + "/" + fileName;
+    for (int i = 0; i < connectionCounter; i++) {
+        pthread_join(threadA[i], NULL);
+        std::cout << "service Threads joined" << std::endl;
+    }
 
-            int fileSize = atoi(size.c_str());
-            while (true) {
-                if (Utils::fileExists(fullFilePath)) {
-                    while (Utils::getFileSize(fullFilePath) < fileSize) {
-                        bzero(data, INSTANCE_DATA_LENGTH + 1);
-                        read(connFd, data, INSTANCE_DATA_LENGTH);
-                        line = (data);
+    pthread_mutex_destroy(&file_lock);
+}
 
-                        if (line.compare(JasmineGraphInstanceProtocol::FILE_RECV_CHK) == 0) {
-                            write(connFd, JasmineGraphInstanceProtocol::FILE_RECV_WAIT.c_str(),
-                                  JasmineGraphInstanceProtocol::FILE_RECV_WAIT.size());
-                        }
-                    }
-                    break;
-                } else {
-                    sleep(1);
-                    continue;
-                }
-            }
+int deleteGraphPartition(std::string graphID, std::string partitionID) {
+    int status = 0;
+    string partitionFilePath = Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder") + "/" +
+                               graphID + +"_" + partitionID;
+    status |= Utils::deleteDirectory(partitionFilePath);
+    string centalStoreFilePath = Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder") + "/" +
+                                 graphID + +"_centralstore_" + partitionID;
+    status |= Utils::deleteDirectory(centalStoreFilePath);
+    string centalStoreDuplicateFilePath =
+        Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder") + "/" + graphID +
+        +"_centralstore_dp_" + partitionID;
+    status |= Utils::deleteDirectory(centalStoreDuplicateFilePath);
+    string attributeFilePath = Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder") + "/" +
+                               graphID + +"_attributes_" + partitionID;
+    status |= Utils::deleteDirectory(attributeFilePath);
+    string attributeCentalStoreFilePath =
+        Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder") + "/" + graphID +
+        +"_centralstore_attributes_" + partitionID;
+    status |= Utils::deleteDirectory(attributeCentalStoreFilePath);
+    if (status == 0)
```

Comment:
Lets use curly braces for the if else structure. Even thought its not essential it improves the readability.

Reply:
Fixed in next PR

Reply:
Ok